aptos-move/mvhashmap/src/versioned_delayed_fields.rs

```diff
@@ -497,7 +497,9 @@ impl<K: Eq + Hash + Clone + Debug + Copy> VersionedDelayedFields<K> {
             ));
         }
 
-        let mut derived_ids = Vec::new();
+        // Track separately, todo_deltas need to be done before todo_derived
+        let mut todo_deltas = Vec::new();
+        let mut todo_derived = Vec::new();
 
         for id in ids {
             let mut versioned_value = self
```

Comment:
so this was the first one we are holding, and then we were grabbing the other one inside, right? and the problem is inner and outer can be on the same lock shard.. So I checked again and it seems we never hold two implicit locks at the same time. read_checked grabs it but it should go out of scope (there is even a comment, but maybe we should explicitly scope it, have a look).

Reply:
yes I was keeping mutable lock on the id here, and then grabbing a new one on base_id as well